docs/classes.md

```diff
@@ -0,0 +1,428 @@
+    instance = {}
+    instance.walk = sub(meters as integer)
+        print "Walked " + meters.ToStr() + " meters"
+    end sub
+    return instance
+end function
+function Duck(name as string)
+    instance = __Duck_Build()
+    return instance
+end function
+
+function __BabyDuck_Build(name as string, age as integer)
+    instance = __Duck_Build()
+    instance.super0_Walk = instance.walk
+    instance.walk = sub(meters as integer)
+        print "Tripped"
+        m.super0_walk(meters)
+    end sub
+end function
+function BabyDuck(name as string, age as integer)
+    instance = __BabyDuck_Build()
+    return instance
+end function
+```
+</details>
+
+## Public by default
+Class fields and methods are public by default, which aligns with the general BrightScript approach that "everything is public". 
+
+```vb
+class Person
+    'defaults to public
+    name as string
+
+    'specified private
+    private socialSecurityNumber
+
+    'defaults to public
+    function getName()
+        return m.name
+    end function
+
+    'specified private
+    private sub setSocialSecurityNumber(value as string)
+        m.socialSecurityNumber = value
+    end sub
+end class
+```
+<details>
+  <summary>View the transpiled BrightScript code</summary>
+  
+```vb
+function Person()
+    instance = {}
+    'defaults to public
+    instance.name = invalid,
+    
+    'specified private
+    instance.socialSecurityNumber = invalid
+    
+    'defaults to public
+    instance.getName = function()
+        return m.name
+    end function
+
+    'specified private
+    instance.setSocialSecurityNumber = sub(value as string)
+        m.socialSecurityNumber = value
+    end sub
+    return instance
+end function
+```
+</details>
+
+You will get compile-time errors whenever you access private members of a class from outside the class. However, be aware that this is only a compile-time restriction. At runtime, all members are public.
+
+## Dynamic type by default
+You can specify a type for class fields and a return type for methods. However, this is entirely optional. All fields and methods have a default type of `dynamic`. However, BrighterScript will attempt to infer the type from usage. Take this for example:
+
+```vb
+class Person
+    'defaults to type "dynamic"
+    name
+
+    'infers type "integer"
+    age = 12
+
+    'infers type "integer"
+    function getAge()
+        return m.age
+    end function
+end class
+```
+<details>
+  <summary>View the transpiled BrightScript code</summary>
+  
+```vb
+function Person()
+    instance = {}
+
+    'defaults to type "dynamic"
+    instance.name = invalid
+
+    'infers type "integer"
+    instance.age = 12
+
+    'infers type "integer"
+    instance.getAge = function()
+        return m.age
+    end function
+    return instance
+end function
+```
+</details>
+
+## Usage
+In order to use a class, you need to construct one. Based on our person class above, you can create a new person like this:
```

Comment:
worth explicitly calling out the new `new` keyword?

Reply:
That's already explained in this section: https://github.com/rokucommunity/brighterscript/pull/40/files#diff-3031a75b5d15117a376b3869ba2a5b24R143

But I could probably add a little better documentation around it, or maybe declare that section higher up?